File: disappearNumber.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDisappearedNumbers = function (nums) {
  const res = [];
  const arr = [];

  for (const number of nums) {
    if (arr[number] !== true) {
      arr[number] = true;
    }
  }

  for (i = 1; i < nums.length + 1; i++) {
    if (!arr[i]) res.push(i);
  }

  return res;
};
// var findDisappearedNumbers = function (nums) {
//   const map = new Map();
//   const res = [];

//   for (const number of nums) {
//     if (!map.get(number)) {
//       map.set(number, true);
//     }
//   }

//   for (let i = 1; i < nums.length + 1; i++) {
//     if (!map.has(i)) res.push(i);
//   }
//   return res;
// };

File: disappearNumber.py
from typing import List


class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        dictionary = {}
        res = []
        for number in nums:
            dictionary[number] = True

        for i in range(1, len(nums)+1):
            if dictionary.get(i) is None:
                res.append(i)
        return res
